parts/linux/cloud-init/artifacts/ensure_imds_restriction.sh

```diff
@@ -2,17 +2,17 @@
 # Ensure IMDS restriction rule is correctly set on the node, this script will be registered as kubelet service's ExecStart
 # to ensure desired iptables rule will be ensured after the node is restarted.
 # Below 3 variables are passed from kubelet service's drop-in file.
-primaryNicIP="${PRIMARY_NIC_IP}"
-insertRuleToMangleTable="${INSERT_IMDS_RESTRICTION_RULE_TO_MANGLE_TABLE}"
-enableIMDSRestriction="${ENABLE_IMDS_RESTRICTION}"
+primaryNicIP="${PRIMARY_NIC_IP:-""}"
+insertRuleToMangleTable="${INSERT_IMDS_RESTRICTION_RULE_TO_MANGLE_TABLE:-false}"
+enableIMDSRestriction="${ENABLE_IMDS_RESTRICTION:-false}"
 
 echo "Primary NIC IP: $primaryNicIP"
 echo "Insert IMDS restriction rule to mangle table: $insertRuleToMangleTable"
 echo "Enable IMDS restriction: $enableIMDSRestriction"
 
 # If any required variable is not set, exit with error
-if [[ -z "$primaryNicIP" || -z "$insertRuleToMangleTable" || -z "$enableIMDSRestriction" ]]; then
-    echo "One or more required variables are not set, exiting..."
+if [ -z "$primaryNicIP" ]; then
```

Comment:
Just to double check: This is expected to work, because, unlike `ENABLE_IMDS_RESTRICTION` and `INSERT_IMDS_RESTRICTION_RULE_TO_MANGLE_TABLE`, `PRIMARY_NIC_IP` is actually computed/passed in by other scripts, right?

Reply:
correct, this is resolved by CSE at provisioning time (via IMDS)